enterprise_access/apps/api/v1/views/subsidy_access_policy.py

```diff
@@ -851,3 +857,152 @@ def allocate(self, request, *args, **kwargs):
+            return Response("Policy not found", status.HTTP_404_NOT_FOUND)
+
+        # IMPLICITLY ASSUME there is exactly one group associated with the policy
+        # as of 04/2024
+        if not group_uuid:
+            policy_group_association = PolicyGroupAssociation.objects.filter(subsidy_access_policy=policy.uuid).first()
+            if not policy_group_association:
+                return Response("Policy group not found associated with subsidy", status.HTTP_404_NOT_FOUND)
+            group_uuid = policy_group_association.enterprise_group_uuid
+
+        # Request learner aggregate data from the subsidy service for this particular subsidy/policy
+        subsidy_learner_aggregate_dict = get_and_cache_subsidy_learners_aggregate_data(
+            policy.subsidy_uuid,
+            policy.uuid
+        )
+
+        # Request the group member data from platform
+        member_response = LmsApiClient().fetch_group_members(
+            group_uuid=group_uuid,
+            sort_by=request_serializer.validated_data.get('sort_by'),
+            user_query=request_serializer.validated_data.get('user_query'),
+            fetch_removed=request_serializer.validated_data.get('fetch_removed'),
+            is_reversed=request_serializer.validated_data.get('is_reversed'),
+            traverse_pagination=traverse_pagination,
+            page=page,
+        )
+        member_results = member_response.get('results')
+        # Sift through the group members data, zipping the aggregate data from the subsidy service into
+        # each member record, assume enrollment count is 0 if subsidy enrollment aggregate data does not exist.
+        for key, result in enumerate(member_results):
+            enrollment_count = 0
+            if lms_user_id := result.get('lms_user_id'):
+                enrollment_count = subsidy_learner_aggregate_dict.get(lms_user_id, 0)
+            result['enrollment_count'] = enrollment_count
+            member_results[key] = result
+        member_response['results'] = member_results
+
+        # return in a csv format if indicated by query params
+        if request_serializer.validated_data.get('format_csv', False):
+            request.accepted_renderer = GroupMembersWithAggregatesCsvRenderer()
+            request.accepted_media_type = GroupMembersWithAggregatesCsvRenderer().media_type
+            return Response(list(member_results), status=status.HTTP_200_OK, content_type='text/csv')
+
+        # Since we are essentially forwarding all request params to platform, we only need to replace the `next` and
+        # `previous` url values from the response returned by platform to construct a valid response object for the
+        # requester.
+        current_url = request.build_absolute_uri()
+        if not traverse_pagination:
+            if member_response.get('next'):
+                member_response['next'] = current_url.replace(f"page={page}", f"page={str(int(page) + 1)}")
```

Comment:
nit: it's possible that the caller provided neither `page` nor `traverse_pagination` in their request. In which case this `replace()` won't do anything.

Reply:
good point, with the https://github.com/openedx/enterprise-access/pull/446#discussion_r1583793562 change we should be covered though, correct?